### PR TITLE
Script on orchestrator

### PR DESCRIPTION
**Pull request by schaubl**
Connected to #494 

Require:
- slipstream/SlipStreamUI#479
- slipstream/SlipStreamClient#210

### CODE REVIEW

File: jar-persistence/src/main/java/com/sixsq/slipstream/persistence/DeploymentModule.java [hunk 1]
```diff
@@ -35,13 +32,14 @@
 import javax.persistence.OneToMany;
 
 import org.hibernate.annotations.CollectionType;
+import org.simpleframework.xml.ElementList;
```
Comment (konstan) [inline jar-persistence/src/main/java/com/sixsq/slipstream/persistence/DeploymentModule.java]: Unused import.

**Comment by konstan** [comment]
I deployed from the two branches (with the changes for server and client).  Tried as test the system-testing deployment.  The script on Orch gets run.  Two problems though.

1. (minor) - `statecustom` is not displayed in `jit` on orchestrator.
2. started mutable run with 0 multiplicity for `apache` and `testclient` (orch. ran the exec script), then attempted to scale the `testclient` but got 500
```
Caused by: java.lang.NullPointerException
        at com.sixsq.slipstream.factory.DeploymentFactory.addParameterMapping(DeploymentFactory.java:273)
        at com.sixsq.slipstream.run.RunNodeResource.createNodeInstanceRuntimeParameters(RunNodeResource.java:292)
        at com.sixsq.slipstream.run.RunNodeResource.createNodeInstanceOnRun(RunNodeResource.java:276)
        at com.sixsq.slipstream.run.RunNodeResource.addNodeInstancesInTransaction(RunNodeResource.java:142)
        at com.sixsq.slipstream.run.RunNodeResource.addNodeInstances(RunNodeResource.java:113)
        ... 79 more
2015-12-01T16:57:15.657+0000 INFO org.restlet.engine.log.LogFilter afterHandle 2015-12-01       16:57:15        127.0.0.1       test    127.0.0.1       8182    POST    /run/a19b2d24-4385-4f14-8ea5-a61f1c2f2e53/testclient    -       500     339     -       69      https://localhost:8182  Python-httplib2/0.9.2 (gzip)    -
```

I don't know yet if this is related to the changes you've made.  I'll be looking into this later.

I haven't reviewed the code yet either.

**Comment by konstan** [comment]
This is the test instance https://185.19.29.113 (super:0dc401848ac1,test:8c793f6c4a7c) from this deployment https://nuv.la/run/48f19c4f-61c3-4826-bd94-8abdb3e04051

**Comment by konstan** [comment]
`2.` Is not a regression.  It's not working on SS v2.19.1 either.  https://github.com/slipstream/SlipStreamServer/issues/565

**Comment by schaubl** [comment]
`1.` Yes, you're right. I think it's not critical for know and we can keep it like this. Do you agree ?

**Comment by konstan** [comment]
`1.` Agreed.

**Comment by konstan** [comment]
Lets try to find a better name for `TargetedModule`.

Also, I want to suggest the following change (if we don't find a better name for the above class).

`TargetedFormProcessor -> TargetedModuleFormProcessor`

**Comment by konstan** [comment]
Proposed change (thanks to @loomis)

`TargetedModule  -> TargetContainerModule`

`TargetedFormProcessor -> TargetContainerModuleFormProcessor`

**Comment by schaubl** [comment]
:+1:

**Comment by konstan** [comment]
Updating on my side.  Will be pushing shortly.  Now checking two NPEs (which are highly likely not related to the changes in this PR.)

```
INFO: VmRuntimeParameterMapping loaded by id = com.sixsq.slipstream.persistence.VmRuntimeParameterMapping@3ed887ee
java.lang.NullPointerException
        at com.sixsq.slipstream.connector.ConnectorBase.getMachineImageLoginUsername(ConnectorBase.java:465)
        at com.sixsq.slipstream.connector.ConnectorBase.getLoginUsername(ConnectorBase.java:454)
        at com.sixsq.slipstream.connector.ConnectorBase.getSshUrl(ConnectorBase.java:214)
        at com.sixsq.slipstream.connector.ConnectorBase.updateMachineInstanceUrlSshOnRun(ConnectorBase.java:206)
        at com.sixsq.slipstream.connector.ConnectorBase.updateInstanceIdAndIpOnRun(ConnectorBase.java:153)
        at com.sixsq.slipstream.connector.ConnectorBase.updateInstanceIdAndIpOnRun(ConnectorBase.java:140)
        at com.sixsq.slipstream.connector.local.LocalConnector.launch(LocalConnector.java:133)
        at com.sixsq.slipstream.connector.Launcher$SyncLauncher.launchRun(Launcher.java:97)
        at com.sixsq.slipstream.connector.Launcher$SyncLauncher.launch(Launcher.java:81)
        at com.sixsq.slipstream.connector.Launcher$SyncLauncher.run(Launcher.java:73)
        at com.sixsq.slipstream.connector.Launcher.launch(Launcher.java:45)
        at slipstream.async.launcher$launch_BANG_$fn__5830$state_machine__3940__auto____5831$fn__5833.invoke(launcher.clj:51)
        at slipstream.async.launcher$launch_BANG_$fn__5830$state_machine__3940__auto____5831.invoke(launcher.clj:51)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:1011)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:1015)
        at slipstream.async.launcher$launch_BANG_$fn__5830.invoke(launcher.clj:51)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

```
Dec 02, 2015 1:58:48 PM com.sixsq.slipstream.connector.local.LocalConnector launch
INFO:   Type: Orchestration
java.lang.NullPointerException
        at com.sixsq.slipstream.connector.local.LocalConnector.launchDeployment(LocalConnector.java:157)
        at com.sixsq.slipstream.connector.local.LocalConnector.launch(LocalConnector.java:121)
        at com.sixsq.slipstream.connector.Launcher$SyncLauncher.launchRun(Launcher.java:97)
        at com.sixsq.slipstream.connector.Launcher$SyncLauncher.launch(Launcher.java:81)
        at com.sixsq.slipstream.connector.Launcher$SyncLauncher.run(Launcher.java:73)
        at com.sixsq.slipstream.connector.Launcher.launch(Launcher.java:45)
        at slipstream.async.launcher$launch_BANG_$fn__5830$state_machine__3940__auto____5831$fn__5833.invoke(launcher.clj:51)
        at slipstream.async.launcher$launch_BANG_$fn__5830$state_machine__3940__auto____5831.invoke(launcher.clj:51)
        at clojure.core.async.impl.ioc_macros$run_state_machine.invoke(ioc_macros.clj:1011)
        at clojure.core.async.impl.ioc_macros$run_state_machine_wrapped.invoke(ioc_macros.clj:1015)
        at slipstream.async.launcher$launch_BANG_$fn__5830.invoke(launcher.clj:51)
        at clojure.lang.AFn.run(AFn.java:22)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Pull request merged**